v1/client.go

```diff
@@ -11,6 +11,9 @@ import (
 //
 // Clients are backed by a gRPC client and as such are thread-safe.
 type Client struct {
+	// Provide a handle on the underlying connection to enable cleanup
+	// behaviors (among others)
+	Conn *grpc.ClientConn
```

Comment:
let's make this `conn` (private) and then add a top level `func (Client) Close() error` method?